chunk/chunk_store.go

```diff
@@ -104,47 +125,132 @@ func init() {
 type Store interface {
 	Put(ctx context.Context, chunks []Chunk) error
 	Get(ctx context.Context, from, through model.Time, matchers ...*metric.LabelMatcher) ([]Chunk, error)
+	Stop()
 }
 
 // StoreConfig specifies config for a ChunkStore
 type StoreConfig struct {
 	S3URL       string
 	DynamoDBURL string
 	ChunkCache  *Cache
+
+	// Not exported as only used by tests to inject mocks
+	dynamodb dynamodbClient
+	s3       s3Client
```

Comment:
I'll do a techdebt ticket to split this up; we basically have the beginning of our own dynamodb client now.